client-tools/Shell/ClientFeature.cpp

```diff
@@ -125,11 +125,13 @@ void ClientFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
 arangosh without connecting to a server.)");
   }
 
-  options->addOption("--server.password",
-                     "The password to use when connecting. If not specified "
-                     "and authentication is required, the user is prompted for "
-                     "a password",
-                     new StringParameter(&_password));
+  options->addOption(
+      "--server.password",
+      "The password to use when connecting. If not specified and "
+      "authentication is required, you are prompted for a password. "
+      "You can wrap the name of an environment variable in at signs to use its "
+      "value, like @ARANGO_PASSWORD@. Literal @ need to be escaped as @@.",
```

Comment:
This may be slightly misleading. It is true that literal `@` need to be escaped, but only for passwords that are specified via the command-line option, and not in the password prompt.